book/src/development/defining_lints.md

```diff
@@ -0,0 +1,199 @@
+# Define New Lints
+
+The first step in the journey of a new lint is the definition
+and registration of the lint in Clippy's codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+#### Lint types
+
+A lint type is the category of items and expressions in which your lint focuses on.
+
+As of the writing of this documentation update, there are 12 groups (a.k.a. _types_)
```

Comment:
```suggestion
As of the writing of this documentation update, there are 12 _types_
```

I would just consistently call it lint types to avoid all confusion with lint groups.